common/template_string.h

```diff
@@ -0,0 +1,79 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_COMMON_TEMPLATE_STRING_H_
+#define CARBON_COMMON_TEMPLATE_STRING_H_
+
+#include "llvm/ADT/StringRef.h"
+
+namespace Carbon {
+
+// Represent a compile-time string in a form suitable for non-type template
+// arguments.
+//
+// These arguments are required to be "structural", and so we copy the string
```

Comment:
Until I looked up "structural" and found https://en.cppreference.com/w/cpp/language/template_parameters I was pretty confused by this.

Reply:
Added a link to the specific subsection of that article.